redpanda/values.yaml

```diff
@@ -163,9 +161,108 @@ listeners:
+    # This should be true unless the container has CPU affinity (eg. min and max above are equal).
+    # Equivalent to: --idle-poll-time-us 0 --thread-affinity 0 --poll-aio 0
+    # overprovisioned: false
+  #
+  memory:
+    # Enables memory locking.
+    # For production: true
+    # enable_memory_locking: false
+    #
+    # It is recommended to have at least 2Gi of memory per core for the Redpanda binary.
+    # This memory is taken from the total memory given to each container.
+    # We allocate 80% of the container's memory to Redpanda, leaving the rest for
+    # the Seastar subsystem (reserveMemory) and other container processes.
+    # So at least 2.5Gi per core is recommended in order to ensure Redpanda has a full 2Gi.
+    #
+    # These values affect --memory and --reserve-memory flags passed to Redpanda and the memory
+    # requests/limits in the StatefulSet.
+    # Valid suffixes: k M G Ki Mi Gi
+    # Only support a single decimal (eg. 2.5Gi rather than 2.55Gi)
+    #
+    container:
+      # Minimum memory count for each Redpanda broker
+      # If omitted, the min value will equal the max value (requested resources defaults to limits)
+      # Equivalent to: resources.requests.memory
+      # For production: 10Gi or greater
+      # min: 2.5Gi
+      #
+      # Minimum memory count for each Redpanda broker
+      # Equivalent to: resources.limits.memory
+      # For production: 10Gi or greater
+      max: 2.5Gi
+    #
+    # redpanda:
+      # This optional redpanda section allows specifying the memory size for both the Redpanda
+      # process and the underlying reserved memory (used by Seastar).
+      # This section is omitted by default, and memory sizes are calculated automatically
+      # based on container memory.
+      # Uncommenting this section and setting memory and reserveMemory values will disable
+      # automatic calculation.
+      #
+      # If you are setting the following values manually, keep in mind the following guidelines (getting
+      # this wrong will potentially lead to performance issues, instability, loss of data, etc.):
+      # The amount of memory to allocate to a container is determined by the sum of three values:
+      # 1. Redpanda (at least 2Gi per core, ~80% of the container's total memory)
+      # 2. Seastar subsystem (200Mi * 0.2% of the container's total memory, 200Mi < x < 1Gi)
+      # 3. other container processes (whatever small amount remains)
+      #
+      # Memory for the Redpanda process.
+      # This must be lower the container's memory (resources.memory.container.min if provided, otherwise
+      # resources.memory.container.max).
+      # Equivalent to: --memory
+      # For production: 8Gi or greater
+      # memory: 2Gi
+      #
+      # Memory reserved for the Seastar subsystem.
+      # Any value above 1Gi will provide diminishing performance benefits.
+      # Equivalent to: --reserve-memory
+      # For production: 1Gi
+      # reserveMemory: 200Mi
```

Comment:
All memory values, including these which go to Seastar, are in base-2 digits and also use the "correct" base-2 suffix (Ki, Mi, Gi). This is change from how Seastar handles memory values underneath. These values here are passed through the new helm template `redpanda-memoryToMi` to appropriately handle the base-2 suffix given here. The output is a digit, which is then combined with Seastar's expected suffix `M` in `configmap.yaml`.